stl/inc/format

```diff
@@ -0,0 +1,2734 @@
+            _Fill_right = _Specs._Width - _Width - _Fill_left;
+            break;
+        case _Align::_None:
+            _STL_ASSERT(false, "Invalid alignment");
+            break;
+        }
+    }
+
+    const basic_string_view<_CharT> _Fill_char{_Specs._Fill, _Specs._Fill_length};
+    for (; _Fill_left > 0; --_Fill_left) {
+        _Out = _RANGES copy(_Fill_char, _STD move(_Out)).out;
+    }
+
+    _Out = _Fn(_STD move(_Out));
+
+    for (; _Fill_right > 0; --_Fill_right) {
+        _Out = _RANGES copy(_Fill_char, _STD move(_Out)).out;
+    }
+
+    return _Out;
+}
+
+template <integral _Integral>
+_NODISCARD constexpr string_view _Get_integral_prefix(const char _Type, const _Integral _Value) {
+    switch (_Type) {
+    case 'b':
+        return "0b"sv;
+    case 'B':
+        return "0B"sv;
+    case 'x':
+        return "0x"sv;
+    case 'X':
+        return "0X"sv;
+    case 'o':
+        if (_Value != _Integral{0}) {
+            return "0"sv;
+        }
+        return {};
+    default:
+        return {};
+    }
+}
+
+template <class _OutputIt>
+_NODISCARD _OutputIt _Write_sign(_OutputIt _Out, const _Sign _Sgn, const bool _Is_negative) {
+    if (_Is_negative) {
+        *_Out++ = '-';
+    } else {
+        switch (_Sgn) {
+        case _Sign::_Plus:
+            *_Out++ = '+';
+            break;
+        case _Sign::_Space:
+            *_Out++ = ' ';
+            break;
+        case _Sign::_None:
+        case _Sign::_Minus:
+            break;
+        }
+    }
+    return _Out;
+}
+
+inline void _Buffer_to_uppercase(char* _Begin, const char* _End) {
+    for (; _Begin != _End; ++_Begin) {
+        *_Begin = static_cast<char>(_CSTD toupper(*_Begin));
+    }
+}
+
+template <class _CharT, integral _Ty>
+_NODISCARD constexpr bool _In_bounds(const _Ty _Value) {
+    if constexpr (is_unsigned_v<_CharT> && is_unsigned_v<_Ty>) {
+        return _Value <= (numeric_limits<_CharT>::max)();
+    } else if constexpr (is_unsigned_v<_CharT>) {
+        return _Value >= 0 && static_cast<make_unsigned_t<_Ty>>(_Value) <= (numeric_limits<_CharT>::max)();
+    } else if constexpr (is_unsigned_v<_Ty>) {
+        return _Value <= static_cast<make_unsigned_t<_CharT>>((numeric_limits<_CharT>::max)());
+    } else {
+        return (numeric_limits<_CharT>::min)() <= _Value && _Value <= (numeric_limits<_CharT>::max)();
+    }
+}
+
+_NODISCARD inline int _Count_separators(size_t _Digits, const string_view _Groups) {
+    int _Separators = 0;
+    if (!_Groups.empty()) {
+        // Calculate the amount of separators that are going to be inserted based on the groupings of the locale.
+        auto _Group_it = _Groups.begin();
+        while (_Digits > static_cast<size_t>(*_Group_it)) {
+            _Digits -= static_cast<size_t>(*_Group_it);
+            ++_Separators;
+            if (_Group_it + 1 != _Groups.end()) {
+                ++_Group_it;
+            }
+        }
+    }
+    return _Separators;

```

Comment:
I believe it would be cleaner if we would directly return for empty groups
```suggestion
    if (_Groups.empty()) {
        return 0;
    }
    
    // Calculate the amount of separators that are going to be inserted based on the groupings of the locale.
    int _Separators = 0;
    auto _Group_it = _Groups.begin();
    while (_Digits > static_cast<size_t>(*_Group_it)) {
        _Digits -= static_cast<size_t>(*_Group_it);
        ++_Separators;
         if (_Group_it + 1 != _Groups.end()) {
            ++_Group_it;
        }
    }
    }
    return _Separators;
```